### PR TITLE
Finish converting Bril programs to Tree Assume, with testing

### PR DESCRIPTION
**Pull request by oflatt**
This PR finishes the conversion to the tree assume encoding, and tests running them against the interpreter.
The main thing it adds is conversion of if and switch statements. Besides that, it fixes some bugs and adds a couple more binary operators to the encoding.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -64,18 +68,63 @@ impl Optimizer {
...
+        match program {
+            Interpretable::Bril(program) => Self::interp_bril(program, args, profile_out),
+            Interpretable::TreeProgram(program) => {
+                let (val, mut printed) = interpret_tree_prog(program, Self::parse_arguments(args));
+                assert_eq!(val, Value::Tuple(vec![]));
+                // add new line to the end of each line in printed
+                for line in printed.iter_mut() {
+                    line.push('\n');
+                }
+                printed.join("")
```
Comment (ezrosent) [inline src/lib.rs:107]: is this different than let mut res = printed.join("\n"); printed.push('\n'); ?

File: src/rvsdg/tree_unique/to_tree.rs [hunk 2]
```diff
@@ -202,8 +202,112 @@ impl<'a> RegionTranslator<'a> {
...
+                    else_branch,
+                } => {
+                    let mut input_values = vec![];
+                    for input in inputs {
+                        input_values.push(self.translate_operand(*input));
+                    }
+                    let pred = self
+                        .translate_operand(*pred)
+                        .to_expr()
+                        .unwrap_or_else(|| panic!("Pred was a state edge"));
```
Comment (ezrosent) [inline src/rvsdg/tree_unique/to_tree.rs:218]: expect?

File: src/rvsdg/tree_unique/to_tree.rs [hunk 3]
```diff
@@ -285,12 +390,40 @@ impl<'a> RegionTranslator<'a> {
...
+                );
+                let mut ret_values = vec![];
+                if output_type.is_some() {
+                    ret_values.push(ArgOrState::Arg(self.current_num_args));
+                    self.current_num_args += 1;
+                }
+                let num_state_edges = num_ret_values - output_type.is_some() as usize;
+
+                // push a state edge for every extra return value
+                for _ in 0..num_state_edges {
```
Comment (ezrosent) [inline src/rvsdg/tree_unique/to_tree.rs:419]: optional: `ret_values.resize_with(ret_values.len()+num_state_edges, || ArgOrState::StateEdge)`

**Review by ezrosent — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (oflatt) [reply src/lib.rs:107]: Yep- in the case that there is nothing printed

**Pull request merged**